pkg/cluster/storageSuffix.go

```diff
@@ -0,0 +1,36 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+)
+
+func openShiftClusterDocumentMutatorFn(doc *api.OpenShiftClusterDocument) error {
+	const ImageRegistry = "imageregistry"
+
+	if doc.OpenShiftCluster.Properties.StorageSuffix == "" {
+		var err error
+
+		doc.OpenShiftCluster.Properties.StorageSuffix, err = randomLowerCaseAlphanumericStringWithNoVowels(5)
+		if err != nil {
+			return err
+		}
+
+		doc.OpenShiftCluster.Properties.ImageRegistryStorageAccountName = ImageRegistry + doc.OpenShiftCluster.Properties.StorageSuffix
```

Comment:
nit: we don't need this line as it is duplicated on line 26
```suggestion
```

Reply:
Well spotted. Thanks! Done.